build_tools/github_actions/post_build_upload.py

```diff
@@ -210,6 +211,47 @@ def upload_logs_to_s3(run_id: str, amdgpu_family: str, build_dir: Path):
         log(f"[INFO] No index.html found at {log_dir}. Skipping index upload.")
 
 
+def find_manifest(build_dir: Path) -> Optional[Path]:
+    """
+    Look for therock-manifest.json file in the locations.
+    Return the first match or None.
+    """
+    candidates = [
+        # Generated (build tree)
+        build_dir / "base" / "aux-overlay" / "build" / "therock-manifest.json",
+        # Installed into the aux-overlay staging folder
+        build_dir
+        / "base"
+        / "aux-overlay"
+        / "stage"
+        / "share"
+        / "therock"
+        / "therock-manifest.json",
+    ]
+    for p in candidates:
+        if p.is_file():
+            return p
+    return None
+
+
+def upload_manifest_to_s3(run_id: str, amdgpu_family: str, build_dir: Path):
+    """
+    Upload therock-manifest.json to:
+      s3://<bucket>/<external_repo_path><run_id>-<platform>/manifests/<amdgpu_family>/therock-manifest.json
+    """
+    external_repo_path, bucket = retrieve_bucket_info()
+    bucket_uri = f"s3://{bucket}/{external_repo_path}{run_id}-{PLATFORM}"
```

Comment:
We should cache the results of `retrieve_bucket_info()` (either internally in that function or here at the call sites). It now makes some API calls that are potentially expensive (network upload/download, API rate limits).

I see it called 4 times here: https://github.com/ROCm/TheRock/actions/runs/18603952466/job/53048947193?pr=1790#step:17:3383